include/envoy/thread/thread.h

```diff
@@ -18,6 +37,8 @@ class Thread {
    * Join on thread exit.
    */
   virtual void join() PURE;
+
+  virtual ThreadHandle handle() const PURE;
```

Comment:
Do we need to expose the thread handle? I don't see this method getting called anywhere.

Reply:
We have some code in our fork (specifically the Windows version of Filesystem::WatcherImpl) that needs access to the thread handle. We haven't PR'd those changes back upstream yet but plan to eventually

Reply:
For this PR can we remove the handle, as it's not used yet?

Then in the next PR we can discuss the handle and its usage in FilesystemImpl. I predict that I'm would suggest you make a windows-specific subclass of that too, but can't tell until I see it :)

In general I hope we can approach portability with injection of alternate class implementations rather than ifdefs, particularly ifdefs in an interface header.

Reply:
Not a problem. 

The specific usage is our workaround for the fact that libevent can only watch sockets on Windows (as opposed to arbitrary). But we'll get to that in a future PR